Best practices/Debugging Source Generator.md

```diff
@@ -0,0 +1,95 @@
+    |:--:|
+    | Figure 3 - Visual Studio modifying window |
+
+2. Add IsRoslynComponent property in source generator project file:
+   
+    ```xml
+    <Project Sdk="Microsoft.NET.Sdk">
+        <PropertyGroup>
+            <TargetFramework>netstandard2.0</TargetFramework>
+            <IsRoslynComponent>true</IsRoslynComponent>
+            <Nullable>enable</Nullable>
+            <LangVersion>latest</LangVersion>
+        </PropertyGroup>
+        <ItemGroup>
+            <PackageReference Include="Microsoft.CodeAnalysis.CSharp" Version="4.0.1" PrivateAssets="all" />
+            <PackageReference Include="Microsoft.CodeAnalysis.Analyzers" Version="3.3.3">
+                <PrivateAssets>all</PrivateAssets>
+                <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+            </PackageReference>
+        </ItemGroup>
+    </Project>
+    ```
+
+3. Add reference to source generator in a project that will be using it (if it already isn't added):
+   
+    ```xml
+    <ItemGroup>
+        <ProjectReference Include="..\JsonApi.SourceGenerator\JsonApi.SourceGenerator.csproj" OutputItemType="Analyzer" ReferenceOutputAssembly="true" />
+    </ItemGroup>
+    ```
+
+4. Add debug configuration using following steps:
+   
+   1. Right click on the project that contains source generator
+   2. Click `Properties`
+   3. Click `Debug`
+   4. Click `Open debug launch profiles UI`
+      | ![debugProperties](/resources/debug-properties.png) |
+      |:--:|
+      | Figure 4 - Debug properties window |
+   5. Delete existing profiles
+      | ![deleteProfile](/resources/delete-existing-profile.png) |
+      |:--:|
+      | Figure 5 - Delete existing profile |
+   6. Add new `Roslyn Component` profile
+      | ![createProfile](/resources/create-roslyn-component-profile.png) |
+      |:--:|
+      | Figure 6 - Add new profile |
+   7. Choose target project from list of projects
+      | ![chooseTargetProject](/resources/choose-target-project.png) |
+      |:--:|
+      | Figure 7 - Choose target project |
+   8. **Optional**: Rename the debug profile so you can easily find and use it later
+
+| ![endResult](/resources/end-result.png) |
+|:--:|
+| Figure 7 - Debug profile after creation |
+
+After these steps you will need to restart Visual Studio and set your source generator project as a startup project. 
+
+In the end add some breakpoints to the source generator code, run the project in debug mode and resolve the issues that were bothering you in the first place or just explore how your source generator works in background.
+
+Sources:
+- [dotnet-how-to-debug-source-generator-vs2022](https://github.com/JoanComasFdz/dotnet-how-to-debug-source-generator-vs2022)
+- [Source Generators](https://docs.microsoft.com/en-us/dotnet/csharp/roslyn-sdk/source-generators-overview)
+- [Debugging C# Source Generators with Visual Studio 2019 16.10](https://stevetalkscode.co.uk/debug-source-generators-with-vs2019-1610)
```

Comment:
I think that the formatting might not work if you don't add an empty space between the title and the bullet points. Usually MD should support this formatting without the , but our handbook as a few unexpected behaviors like this one.